### PR TITLE
Add animations to navbar dropdowns on mobile

### PR DESCRIPTION
**Pull request by Mandrenkov**
**Context:**

The PennyLane website has transition animations when a navbar dropdown is expanded or collapsed on mobile.

**Description of the Change:**

* Added an animation when the mobile navbar dropdown is expanded or collapsed.
* Added an animation when a dropdown from a navbar link on mobile is expanded or collapsed.

**Benefits:**

* There is greater parity between the consumers of the XST and the main PennyLane website.

**Possible Drawbacks:**

None.

**Related GitHub Issues:**

None.

---

**Preview:** https://xanadu-sphinx-theme--45.org.readthedocs.build/en/45/

### CODE REVIEW

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,9 @@
   text and ARIA labels and fixing issues with keyboard navigation.
   [(#43)](https://github.com/XanaduAI/xanadu-sphinx-theme/pull/43)
 
+* Added transition animations to the navbar dropdowns on mobile.
+  [(#44)](https://github.com/XanaduAI/xanadu-sphinx-theme/pull/44)
```
Comment (Mandrenkov) [inline .github/CHANGELOG.md:10]:
```suggestion
  [(#45)](https://github.com/XanaduAI/xanadu-sphinx-theme/pull/45)
```

**Comment by LucasSilbernagel** [comment]
Nice! Could we get a preview link?

**Comment by Mandrenkov** [comment]
> Nice! Could we get a preview link?

Oops! I've added it now. Sorry about that, @LucasSilbernagel !

**Review by LucasSilbernagel — APPROVED** [review]
LGTM, thanks @Mandrenkov !

**Pull request merged**